src/FileStorageQueue/fsq.h

```diff
@@ -413,7 +413,9 @@ class FSQ final : public CONFIG::T_FILE_NAMING_STRATEGY,
       if (next_file) {
         //      const FileProcessingResult result = processor_.OnFileReady(*next_file.get(),
         //      time_manager_.Now());
-        const bool successfully_processed = processor_.OnFileReady(next_file->full_path_name, next_file->size);
+        // AlexZ: we can't believe next_file->size here. Debugging shows that it can be greater than real file size.
+        // TODO: refactor FSQ or capture a bug.
```

Comment:
Ah, wait, you encountered a bug in FSQ? That's scary, let's char. It uses ```stat()``` internally, which is unlikely to go wrong.

We might be appending to a finalized file, or need an explicit `sync` or God knows what.